utils.py

```diff
@@ -0,0 +1,34 @@
+import yaml
+
+def update_args(parser, args):
+    """
+    update_args replaces default values of the args. Arguments that are directly
+    specified from CLI has more priority. 
+    
+    For instance, if the value of an argument has default value, and if the value of 
+    that argument is defined in YAML file, the latter value will be kept. 
+
+    If the value of an argument is specified manually/directly from the CLI (not default),
+    but if the value of that argument is defined in YAML file, the latter value will be 
+    ignored. Hence, the former value will be kept.
+    """
+    if args.from_config is not None:
```

Comment:
Raise an error if it's None, otherwise we don't know what to do and it might lead to unexpected consequences further down the line.

Reply:
the logic here is if there is no config YAML file provided, the default values along with manually/directly input arguments will be used. That is the behaviour when `args.from_config is None`.

I am not sure if I understand correctly, but this coding pattern is somewhat preferred pattern in lots of programming language. However, I am not super confident if it is for Python.

Reply:
Indeed a little strange. The method doesn't tell me the default behaviour what you explained.

Reply:
Going go make the change as you suggested!